publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java

```diff
@@ -198,22 +199,13 @@ public ResponseEntity generateAndPublish(@ApiParam(value = "message protocol", r
         catch (RemRemPublishException e) {
                 return new ResponseEntity(e.getMessage(), HttpStatus.NOT_FOUND);
         }
-        catch (Exception e) {
-            log.info("The result from REMReM Generate is not OK and have value: " + e.getMessage());
-            if (e.getMessage().startsWith(Integer.toString(HttpStatus.BAD_REQUEST.value()))) {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.GENERATE_BAD_REQUEST), HttpStatus.BAD_REQUEST);
-            } else if (e.getMessage().startsWith(Integer.toString(HttpStatus.SERVICE_UNAVAILABLE.value()))) {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.GENERATE_NO_SERVICE_ERROR), HttpStatus.SERVICE_UNAVAILABLE);
-            } else if (e.getMessage().startsWith(Integer.toString(HttpStatus.UNAUTHORIZED.value()))) {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.GENERATE_UNAUTHORIZED), HttpStatus.UNAUTHORIZED);
-            }else if (e.getMessage().startsWith(Integer.toString(HttpStatus.NOT_ACCEPTABLE.value()))) {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.ERLOOKUP_NONEFOUND), HttpStatus.NOT_ACCEPTABLE);
-            }else if (e.getMessage().startsWith(Integer.toString(HttpStatus.EXPECTATION_FAILED.value()))) {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.ERLOOKUP_MULTIPLEFOUND), HttpStatus.EXPECTATION_FAILED);
-            }
-            else {
-                return new ResponseEntity(parser.parse(RemremPublishServiceConstants.GENERATE_INTERNAL_ERROR), HttpStatus.INTERNAL_SERVER_ERROR);
+        catch (HttpStatusCodeException e) {
+            HttpStatus result = HttpStatus.BAD_REQUEST;
+            EnumSet<HttpStatus> getStatus = EnumSet.of(HttpStatus.SERVICE_UNAVAILABLE, HttpStatus.UNAUTHORIZED, HttpStatus.NOT_ACCEPTABLE, HttpStatus.EXPECTATION_FAILED, HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
Please initialize _getStatus_ the class level. 
If it is in catch each request to create _getStatus_.